litestar/contrib/jwt/jwt_token.py

```diff
@@ -82,7 +85,12 @@ def decode(encoded_token: str, secret: str | dict[str, str], algorithm: str) ->
             payload = jwt.decode(token=encoded_token, key=secret, algorithms=[algorithm], options={"verify_aud": False})
             exp = datetime.fromtimestamp(payload.pop("exp"), tz=timezone.utc)
             iat = datetime.fromtimestamp(payload.pop("iat"), tz=timezone.utc)
-            return Token(exp=exp, iat=iat, **payload)
+            field_names = {f.name for f in dataclasses.fields(Token)}
+            extra_fields = set(payload.keys()).difference(field_names)
+            extras = {}
+            for key in extra_fields:
+                extras[key] = payload.pop(key)
+            return Token(exp=exp, iat=iat, **payload, _extras=extras)
```

Comment:
Not really related - but this method looks like it could be a `@classmethod`, typed to return `Self` and `return cls(...)` etc.